src/meta-cli/src/deploy/actors/typegate.rs

```diff
@@ -206,6 +216,14 @@ impl Handler<message::Stop> for TypegateActor {
     type Result = ();
 
     fn handle(&mut self, _msg: message::Stop, ctx: &mut Self::Context) {
+        if let Some(child_arc) = self.child_process.take() {
+            let fut = async move {
+                let mut child = child_arc.lock().await;
+                let _ = child.kill().await;
```

Comment:
Not sure if I understand correctly, but looks like the mutex will be locked in the code above until the child process exits.
> ```
> let mut child = child_arc.lock().await;
> child
>   .wait()
>   .await
>   .expect("failed to wait for typegate process");
> addr.do_send(message::Stop);
> ```

So the lock would only be acquired here after the child process has stopped...

Reply:
Good catch, in fact none of this are required now. The [Signal hook](https://docs.rs/signal-hook/latest/signal_hook/) overrides most of the logic.